File: src/main/java/net/gamershub/kitpvp/enchantments/CustomEnchantment.java
package net.gamershub.kitpvp.enchantments;

import lombok.Getter;
import net.minecraft.ChatFormatting;
import net.minecraft.network.chat.CommonComponents;
import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.EquipmentSlot;
import net.minecraft.world.item.enchantment.Enchantment;
import net.minecraft.world.item.enchantment.EnchantmentCategory;
import org.bukkit.event.Listener;
import org.jetbrains.annotations.NotNull;

@Getter
public class CustomEnchantment extends Enchantment implements Listener {
    protected String id;
    protected int maxLevel;

    public CustomEnchantment(String id, int maxLevel, String displayName, EnchantmentCategory category, EquipmentSlot... equipmentSlots) {
        super(Rarity.COMMON, category, equipmentSlots);
        this.id = id;
        this.maxLevel = maxLevel;
        this.descriptionId = displayName;
    }

    @Override
    public @NotNull Component getFullname(int level) {
        return Component.translatable(this.getDescriptionId())
                .withStyle(ChatFormatting.BLUE)
                .append(CommonComponents.SPACE)
                .append(Component.translatable("enchantment.level." + level));
    }

}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/enchantments/CustomEnchantment.kt
package io.github.chaosdave34.kitpvp.enchantments

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.Material
import org.bukkit.event.Listener
import org.bukkit.inventory.ItemStack


abstract class CustomEnchantment(val id: String, val name: String) : Listener {

    fun getDisplayName(level: Int): Component {
        return Component.text(name, NamedTextColor.BLUE)
            .append(Component.space())
            .append(Component.translatable("enchantment.level.$level"))
    }

    fun ItemStack.containsThisEnchantment(): Boolean {
        if (this.type == Material.AIR) return false
        return EnchantmentHandler.containsEnchantment(this, id)
    }

    fun ItemStack.getEnchantmentLevel(): Int {
        return EnchantmentHandler.getEnchantmentLevel(this, id)
    }
}